### PR TITLE
Resolve conflicts from Main branch

### PR DESCRIPTION
**Pull request by yan-man**
- resolve conflicts, make sure contracts compile properly
- resolve tests later in a different PR

### CODE REVIEW

File: src/contracts/BorrowModule.sol [hunk 1]
```diff
@@ -14,39 +14,51 @@ contract BorrowModule is IBorrowModule {
...
   using SafeERC20 for IERC20;
 
+  // debt balances, fetches indexes from liquidity layer
+
+  // keep collateral configuration
+  // By using BorrowModule, LPs can choose which collaterals are used to borrow their assets
+
+  // keep hooks to be executed by LiquidityHub when there is supply/withdraw actions
+
+  // fetch liquidity from liquidityHub
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: We can remove these comments

File: src/contracts/BorrowModule.sol [hunk 2]
```diff
@@ -14,39 +14,51 @@ contract BorrowModule is IBorrowModule {
...
   address public liquidityHub;
+  address public interestRateStrategy;
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: Borrow module does not need interest rate strategy here. Will be fetched from LH

File: src/contracts/BorrowModule.sol [hunk 3]
```diff
@@ -14,39 +14,51 @@ contract BorrowModule is IBorrowModule {
...
   struct Reserve {
     uint256 id;
     address asset;
-    // uint256 totalDebt;
-    // uint256 lastUpdateIndex;
-    // uint256 lastUpdateTimestamp;
+    uint256 totalDebt;
+    uint256 lastUpdateIndex;
+    uint256 lastUpdateTimestamp;
+    uint256 borrowRate;
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: Please revert this change. We are keeping these commented out, and borrowRate is coming from LH

File: src/contracts/BorrowModule.sol [hunk 4]
```diff
@@ -14,39 +14,51 @@ contract BorrowModule is IBorrowModule {
...
     ReserveConfig config;
   }
 
   struct ReserveConfig {
     uint256 lt;
     uint256 lb; // TODO: liquidationProtocolFee
+    uint256 rf;
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: This will be dealt with on LH side, please revert

File: src/contracts/BorrowModule.sol [hunk 5]
```diff
@@ -14,39 +14,51 @@ contract BorrowModule is IBorrowModule {
...
 
   // reserve id => user address => user data
   mapping(uint256 => mapping(address => UserConfig)) public users;
   // reserve id => reserveData
   mapping(uint256 => Reserve) public reserves;
 
-  constructor(address liquidityHubAddress) {
+  constructor(address liquidityHubAddress, address interestRateStrategyAddress) {
     liquidityHub = liquidityHubAddress;
+    interestRateStrategy = interestRateStrategyAddress;
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: Don't need interestRateStrategy here, will fetch from LH

File: src/contracts/BorrowModule.sol [hunk 6]
```diff
@@ -60,9 +72,13 @@ contract BorrowModule is IBorrowModule {
   }
 
   function getUserDebt(uint256 assetId, address user) external view returns (uint256) {
+    // TODO: Instead use a getter from liquidity hub to get up-to-date user debt (with accrued debt)
+    return _getUserDebt(assetId, user);
+  }
+
+  function _getUserDebt(uint256 assetId, address user) internal view returns (uint256) {
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: I was a fan of getting rid of this intermediary function because it isn't so long. If we do want to keep it, we also need to respect solidity guidelines, and put it after all external functions. But I prefer to not have it

File: src/contracts/BorrowModule.sol [hunk 7]
```diff
@@ -132,14 +151,15 @@ contract BorrowModule is IBorrowModule {
   // TODO: Implement repay, calls liquidity hub restore method
   // TODO: onBehalfOf
   function repay(uint256 assetId, uint256 amount) external {
-    ILiquidityHub(liquidityHub).restore(assetId, amount);
+    ILiquidityHub(liquidityHub).restore(assetId, amount, 0);
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol:133]: Can add a TODO to calculate risk premium and pass here to LH

File: src/contracts/BorrowModule.sol [hunk 8]
```diff
@@ -132,14 +151,15 @@ contract BorrowModule is IBorrowModule {
...
 
     emit Repaid(assetId, msg.sender, amount);
   }
 
   // TODO: Needed?
   function getInterestRate(uint256 assetId) public view returns (uint256) {
-    return ILiquidityHub(liquidityHub).getInterestRate(assetId);
+    // read from state, convert to ray
+    return reserves[assetId].borrowRate * 1e23;
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: Please revert this change, borrow rate will not be stored on BM, instead fetched from LH

File: src/contracts/BorrowModule.sol [hunk 9]
```diff
@@ -185,7 +205,78 @@ contract BorrowModule is IBorrowModule {
     );*/
   }
 
+  // TODO: access control
+  function updateInterestRateStrategy(address newInterestRateStrategy) external {
+    interestRateStrategy = newInterestRateStrategy;
+  }
+
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: Don't need this, will be on LH side

File: src/contracts/BorrowModule.sol [hunk 10]
```diff
@@ -185,7 +205,78 @@ contract BorrowModule is IBorrowModule {
...
+          uint40(reserve.lastUpdateTimestamp),
+          block.timestamp
+        )
+      ) +
+      amount;
+
+    reserve.lastUpdateTimestamp = block.timestamp;
+
+    // TODO: update index
+  }
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: Both these update state and accrue interest will happen on LH, please remove from here

File: src/contracts/LiquidityHub.sol [hunk 11]
```diff
@@ -9,6 +9,7 @@ import {MathUtils} from './MathUtils.sol';
 import {ILiquidityHub} from '../interfaces/ILiquidityHub.sol';
 import {IReserveInterestRateStrategy} from '../interfaces/IReserveInterestRateStrategy.sol';
 import {DataTypes} from '../libraries/types/DataTypes.sol';
+import {IBorrowModule} from '../interfaces/IBorrowModule.sol';
```
Comment (CheyenneAtapour) [inline src/contracts/LiquidityHub.sol]: Unused, please remove

File: src/contracts/BorrowModule.sol (hunk 10 above)
Comment (yan-man) [reply src/contracts/BorrowModule.sol]: will user level interest accrual also happen in the LH? or just spoke level

File: src/contracts/BorrowModule.sol (hunk 6 above)
Comment (yan-man) [reply src/contracts/BorrowModule.sol]: removed for now

File: src/contracts/BorrowModule.sol (hunk 10 above)
Comment (CheyenneAtapour) [reply src/contracts/BorrowModule.sol]: So the shares on the Spoke and LH are exactly 1:1, so we only need the actual accrual to happen on LH side. The Spokes should just keep track of their users' shares, and can then distribute interest accordingly to users.

Side point: Can just think of the LH lending to "users" being the Spokes. Whatever users are using the Spokes are abstracted from the point of view of LH. Spokes will just handle everything related to users on Spoke level

File: src/contracts/BorrowModule.sol [hunk 12]
```diff
@@ -60,31 +60,15 @@ contract BorrowModule is IBorrowModule {
...
     // TODO: Instead use a getter from liquidity hub to get up-to-date user debt (with accrued debt)
-    return
-      u.balance.rayMul(
-        MathUtils.calculateCompoundedInterest(
-          getInterestRate(assetId),
-          uint40(u.lastUpdateTimestamp),
-          block.timestamp
-        )
-      );
+    return 0;
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: I think you can leave the above calc in for now, instead of returning 0, because it serves as a good base. Just needs to be adjusted a bit

File: src/contracts/BorrowModule.sol [hunk 13]
```diff
@@ -60,31 +60,15 @@ contract BorrowModule is IBorrowModule {
...
     // TODO: Instead use a getter from liquidity hub to get up-to-date reserve debt (with accrued debt)
-    return
-      r.totalDebt.rayMul(
-        MathUtils.calculateCompoundedInterest(
-          getInterestRate(assetId),
-          uint40(r.lastUpdateTimestamp),
-          block.timestamp
-        )
-      );
+    return 0;
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: Can leave the above calc in for now, similar to above

File: src/contracts/BorrowModule.sol [hunk 14]
```diff
@@ -115,13 +99,14 @@ contract BorrowModule is IBorrowModule {
     emit Withdraw(assetId, msg.sender, amount);
   }
 
+  // TODO: On behalf of and referral code
   function borrow(uint256 assetId, uint256 amount) external {
-    // TODO: On behalf of and referral code
```
Comment (CheyenneAtapour) [inline src/contracts/BorrowModule.sol]: Small nit: I like having the todos as you do, above the function as you have it, but we just kept them within the function as our convention

File: src/contracts/BorrowModule.sol (hunk 12 above)
Comment (yan-man) [reply src/contracts/BorrowModule.sol]: I removed it because we don't have `u.lastUpdateTimestamp` anymore, but will just use 0 there and leave the rest

File: src/contracts/BorrowModule.sol (hunk 13 above)
Comment (yan-man) [reply src/contracts/BorrowModule.sol]: ditto

File: src/contracts/BorrowModule.sol (hunk 14 above)
Comment (yan-man) [reply src/contracts/BorrowModule.sol]: I am agnostic, so I'll keep that in mind going forward
